regression/pages/ecommerce/redeem_coupon_page.py

```diff
@@ -0,0 +1,241 @@
+            '>.discount-mc-price-group .course-new-price>span'
+        ).text[0]
+        return {
+            'course_price': extract_numerical_value(course_price_str),
+            'benefit_value': extract_numerical_value(benefit_value_str),
+            'discounted_price': extract_numerical_value(discounted_price_str),
+            'benefit_type': self.benefit_type
+        }
+
+    @property
+    def benefit_type(self):
+        """
+        Get benefit type
+        Returns:
+            benefit type:
+        """
+        discount_str = self.q(
+            css='' + self.course_tile + ' .discount-percentage>p').text[0]
+        if '$' in discount_str:
+            return 'Absolute'
+        elif '%' in discount_str:
+            return 'Percentage'
+
+    @property
+    def coupon_expiry_date(self):
+        """
+        Get coupon expiry date
+        Returns:
+            coupon expiry date:
+        """
+        date_string = self.q(
+            css='' + self.course_tile + ' .voucher-valid-until>p').text[0]
+        raw_date = re.search('(?<=until ).+', date_string)
+        raw_date = raw_date.group(0)
+        return convert_date_format(raw_date)
+
+    @property
+    def course_start_date(self):
+        """
+        Get course start date
+        Returns:
+            course start date:
+        """
+        date_string = self.q(
+            css='' + self.course_tile + '>.course-start').text[0]
+        raw_date = re.search('(?<=: ).+', date_string)
+        raw_date = raw_date.group(0)
+        return convert_date_format(raw_date)
+
+    @property
+    def course_image_text(self):
+        """
+        Get course image alt text from course tile
+        Returns:
+            alt text
+        """
+        return self.q(
+            css='.discount-multiple-courses>.box-shadow img').attrs('alt')
+
+    @property
+    def course_ids_list(self):
+        """
+        Return the list of course ids
+        Returns:
+            Course Ids
+        """
+        image_links_list = self.q(
+            css='.image-container>.img-responsive').attrs('src')
+        ids_list = [
+            get_course_ids_from_link(elem) for elem in image_links_list]
+        return ids_list
+
+    def checkout_discount_coupon(self, course_id, target_page):
+        """
+        Click on Checkout button to go to basket page
+        Args:
+            course_id:
+            target_page:
+        """
+        self.q(
+            css='#PurchaseCertificate[data-course-id="' + course_id + '"]'
+        ).click()
+        target_page.wait_for_page()
+
+    def redeem_enrollment(self, target_page):
+        """
+        Click on Checkout button to go to basket page
+        Args:
+             target_page:
+        """
+        self.q(css='#RedeemEnrollment').click()
+        target_page.wait_for_page()
+
+    def display_error_message(self, course_id):
+        """
+        Click on Purchase certificate button to display error message
```

Comment:
But, you can click on the purchase certificate button and not get an error, right? I think you're combining two methods and that you've mis-named this, too. Seems like you're putting a test into a page object.